src/components/HorizontalTabs/HorizontalTabs.tsx

```diff
@@ -0,0 +1,51 @@
+import { SxProps, Theme } from '@mui/material/styles';
+import { ReactNode } from 'react';
+import { HorizontalTab } from './HorizontalTab';
+import {
+  HorizontalTabsContainer,
+  HorizontalTabSize,
+} from './HorizontalTabs.style';
+export interface HorizontalTabItem {
+  value: string;
+  label?: ReactNode;
+  startAdornment?: ReactNode;
+  endAdornment?: ReactNode;
+  disabled?: boolean;
+}
+
+export interface HorizontalTabsProps {
+  tabs: HorizontalTabItem[];
+  onChange: (event: React.SyntheticEvent, newValue: string) => void;
+  onTabClick: (
+    value: string,
+  ) => (event: React.MouseEvent<HTMLDivElement>) => void;
+  value?: string;
+  size?: HorizontalTabSize;
+  sx?: SxProps<Theme>;
+}
+
+export const HorizontalTabs = ({
+  tabs,
+  onChange,
+  onTabClick,
+  value,
+  size = HorizontalTabSize.LG,
+  sx,
+}: HorizontalTabsProps) => {
+  return (
+    <HorizontalTabsContainer value={value} onChange={onChange} sx={sx}>
+      {tabs.map((tab) => (
```

Comment:
@dennyscode let's check here if the tab has a one of these properties `label`, `endAdornment` or `startAdornment` and if not use `null`, otherwise instantiate the `HorizontalTab`